hera_qm/metrics_io.py

```diff
@@ -498,7 +498,7 @@ def _parse_list_of_antpols(input_str):
     for entry in entries:
         ant, pol = entry[1:-1].split(",")
         pol = pol[-2]
-        li.append(tuple((int(ant), str(pol))))
+        li.append(tuple((int(ant), qm_utils._str_to_bytes(str(pol)))))
```

Comment:
Why do you need to wrap this in `_str_to_bytes`? Is this list written out to disk at some point? Why isn't the wrapping done as it's being written in that case?

Reply:
so this, and the next one, didn't _use_ to have this. But i noticed some weird type casting when I was running the test suite. I wish I had a better answer as to why the types were messed up but this was the only way I found to fix it.

Reply:
Okay, I think we should refactor how these objects are handled then. We really should be doing the type conversion at the I/O interface, not buried in the dictionary validation code or list parsing code. I'm of the opinion that while these objects are in memory, they should be `str` in both python2 and python3, so no funny business should happen in these functions.

Reply:
I would definitely prefer that they all remain `str` types too. I'll give this stuff another look.